lib/spack/spack/test/variant.py

```diff
@@ -711,3 +714,10 @@ def test_str(self):
         c['feebar'] = SingleValuedVariant('feebar', 'foo')
         c['shared'] = BoolValuedVariant('shared', True)
         assert str(c) == ' feebar=foo foo=bar,baz foobar=fee +shared'
+
+
+def test_disjoint_set_initialization_errors():
```

Comment:
The packages that have been modified to use `any_combination_of` etc.  test this implicitly in the `test_package_sanity` unit test. Do you prefer to have an explicit unit test here just for `disjoint_sets`?

Reply:
I'm satisfied with what you mention (so I don't require it), but IMO it would still be worthwhile to explicitly test the simple object here. There could be issues with `DisjointSet` and additionally issues with how `DisjointSet` is used by the package object so having the separate tests would help there.